compiler/executable/match_/planner/plan.rs

```diff
@@ -943,7 +949,9 @@ impl ConjunctionPlan<'_> {
                         Inputs::Single([rhs_var.unwrap()]),
                         self.type_annotations,
                     ))
-                } else if constraint.unbound_direction() == Direction::Canonical && Some(sort_variable) != rhs_var {
+                } else if constraint.unbound_direction() == Direction::Canonical && Some(sort_variable) != rhs_var
+                    || Some(sort_variable) == lhs_var
```

Comment:
When testing `as` as `sub`, I started encountering `UnboundInverted` iterate mods, which were not expected.

Reply:
That sounds like a planner issue more than anything!

Reply:
So this is (direction == CANONICAL && soft_var != rhs_var) || Some(soft_var) == lhs_var ?

Reply:
Yeah, the code is correct. I can highlight it if you want to